### PR TITLE
Remove "Telemetry V2" usages

### PR DESCRIPTION
**Pull request by kfaseela**
## Description

Fixes https://github.com/istio/istio.io/issues/15336

## Reviewers

- [ ] Ambient
- [X] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Extensions and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure
- [ ] Localization/Translation

### CODE REVIEW

**Comment by kfaseela** [comment]
@howardjohn I see only these two usages in the current docs, remaining are all on older archived docs or similar.

File: content/en/docs/ops/deployment/performance-and-scalability/index.md [hunk 1]
```diff
@@ -71,7 +71,7 @@ Data plane performance depends on many factors, for example:
 - Number of proxy worker threads
 - Protocol
 - CPU cores
-- Number and types of proxy filters, specifically telemetry v2 related filters.
+- Number and types of proxy filters, specifically telemetry related filters.
```
Comment (craigbox) [inline content/en/docs/ops/deployment/performance-and-scalability/index.md:74]: Can a user customise these?  Should we say something like "Number and type of Envoy filters, including those added by telemetry features" ?

Comment (howardjohn) [inline content/en/docs/ops/deployment/performance-and-scalability/index.md:74]:
```suggestion
- Various proxy features. In particular, telemetry filters (logging, tracing, and metrics) are known to have a moderate impact.
```

**Review by howardjohn — COMMENTED** [review]
LGTM thanks!

**Review by howardjohn — APPROVED** [review]

**Review by craigbox — APPROVED** [review]

**Comment by kfaseela** [comment]
@craigbox can we cherrypick this to 1.22 or this is master only?

**Comment by craigbox** [comment]
Yep, I general think everything that isn't specific to 1.23 should be cherry-picked

**Comment by whitneygriffith** [comment]
Looks great! Thanks for doing this!

**Pull request merged**